sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVRelation.scala

```diff
@@ -246,8 +116,10 @@ object CSVRelation extends Logging {
 private[sql] class CSVOutputWriterFactory(params: CSVOptions) extends OutputWriterFactory {
   override def newInstance(
       path: String,
+      bucketId: Option[Int],
       dataSchema: StructType,
       context: TaskAttemptContext): OutputWriter = {
+    if (bucketId.isDefined) sys.error("csv doesn't support bucketing")
```

Comment:
It should, but today it is still the responsibility of the `OutputWriter` to generate the filename and only some of the implementations can handle appending the bucket id.  This should be done as part of [SPARK-13683](https://issues.apache.org/jira/browse/SPARK-13683).